crates/core/src/models/call.rs

```diff
@@ -0,0 +1,328 @@
+use std::slice::SliceIndex;
+
+use reth_primitives::TransactionSigned;
+use reth_rlp::Decodable;
+use starknet::accounts::Call;
+use starknet_crypto::FieldElement;
+
+use super::ConversionError;
+use crate::models::DataDecodingError;
+
+pub struct Calls(Vec<Call>);
+
+impl From<Calls> for Vec<Call> {
+    fn from(calls: Calls) -> Self {
+        calls.0
+    }
+}
+
+impl From<Vec<Call>> for Calls {
+    fn from(calls: Vec<Call>) -> Self {
+        Self(calls)
+    }
+}
+
+/// TryFrom implementation for account contract calls
+impl TryFrom<Vec<FieldElement>> for Calls {
+    type Error = ConversionError;
+
+    fn try_from(value: Vec<FieldElement>) -> Result<Self, Self::Error> {
+        let calls_len = u32::try_from(value[0]).map_err(|e| ConversionError::ValueOutOfRange(e.to_string()))? as usize;
+        let mut offset = calls_len * 4 + 2;
+
+        let mut calls = vec![];
+        for i in 0..calls_len {
+            let calldata_len =
+                u32::try_from(value[i * 4 + 4]).map_err(|e| ConversionError::ValueOutOfRange(e.to_string()))? as usize;
+            let call = Call {
+                to: value[i * 4 + 2],
+                selector: value[i * 4 + 3],
+                calldata: value[offset..offset + calldata_len].to_vec(),
+            };
+            offset += calldata_len;
+            calls.push(call);
+        }
+        Ok(Calls(calls))
+    }
+}
+
+impl TryFrom<&Calls> for TransactionSigned {
```

Comment:
should be instead tdo a `TryFrom<Call>` and move the selection of the first call in the `impl Calls`? replacing `get_eth_transaction` with either `get_eth_transaction(self, index)` of `get_first_eth_transaction`?

Reply:
That makes a lot more sense yes!

Reply:
following up on your comment for the orphan rules, I'd then say ok to keep a `TryFrom<Calls>`, but we then raise when there is more than one call